File: mock-facility-management/src/main/java/de/vkb/dojo/es/facilityManagement/model/eventState/RoomEventAndState.java
package de.vkb.dojo.es.facilityManagement.model.eventState;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonGetter;
import com.fasterxml.jackson.annotation.JsonProperty;
import de.vkb.dojo.es.facilityManagement.model.event.RoomEvent;
import de.vkb.dojo.es.facilityManagement.model.state.Room;

public class RoomEventAndState {
    private final RoomEvent event;
    private final Room state;

    @JsonCreator
    public RoomEventAndState(
            @JsonProperty("event") RoomEvent event,
            @JsonProperty("state") Room state
    ) {
        this.event = event;
        this.state = state;
    }

    @JsonGetter
    public RoomEvent getEvent() {
        return event;
    }

    @JsonGetter
    public Room getState() {
        return state;
    }
}

File: todo-reservation-micronaut/src/main/kotlin/de/vkb/dojo/es/facilityManagement/model/eventState/RoomEventAndState.kt
package de.vkb.dojo.es.facilityManagement.model.eventState

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import de.vkb.dojo.es.facilityManagement.model.event.RoomEvent
import de.vkb.dojo.es.facilityManagement.model.state.Room

@JsonIgnoreProperties(ignoreUnknown = true)
data class RoomEventAndState(
    val event: RoomEvent,
    val room: Room?
)
